controllers/imagebasedgroupupgrade.go

```diff
@@ -142,6 +150,73 @@ func (r *IBGUReconciler) syncStatusWithCGUs(ctx context.Context, ibgu *ibguv1alp
 	return nil
 }
 
+func clusterRoleToBytes(cr *rbac.ClusterRole) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	rbac.AddToScheme(scheme)
+	s := serializer.NewSerializerWithOptions(serializer.DefaultMetaFactory, scheme, scheme, serializer.SerializerOptions{})
+	gvks, isUnversioned, err := scheme.ObjectKinds(cr)
+	if err != nil {
+		return []byte{}, err
+	}
+	if !isUnversioned && len(gvks) == 1 {
+		cr.TypeMeta = metav1.TypeMeta{
+			Kind:       gvks[0].Kind,
+			APIVersion: gvks[0].GroupVersion().Identifier(),
+		}
+	}
+	var dst bytes.Buffer
+	s.Encode(cr, &dst)
+	return dst.Bytes(), nil
+}
+
+func (r *IBGUReconciler) ensurePermissions(ctx context.Context, ibgu *ibguv1alpha1.ImageBasedGroupUpgrade) error {
+	clusterRole := &rbac.ClusterRole{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "open-cluster-management:klusterlet-work:ibu-role",
+			Labels: map[string]string{
+				"open-cluster-management.io/aggregate-to-work": "true",
+			},
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{"lca.openshift.io"},
+				Resources: []string{"imagebasedupgrades"},
+				Verbs:     []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+			},
+		},
+	}
+	crBytes, err := clusterRoleToBytes(clusterRole)
+	if err != nil {
+		return fmt.Errorf("failed to convert ClusterRole to []bytes: %w", err)
+	}
+	manifestWork := &mwv1.ManifestWork{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "",
+			Name:      "ibu-permission-set",
+		},
+		Spec: mwv1.ManifestWorkSpec{
+			Workload: mwv1.ManifestsTemplate{
+				Manifests: []mwv1.Manifest{{RawExtension: runtime.RawExtension{Raw: crBytes}}},
+			},
+		},
+	}
+	for _, cluster := range ibgu.Spec.Clusters {
```

Comment:
Is it cleaner to create this as a mwrs and just include it in the CGU?

Reply:
updated

Reply:
`clusters` actually doesn't include anything if the cluster selector is used. Looks like the only way is to create a mwrs and include it in all CGUs.

Reply:
I was thinking about re using the `getclustersforupgrade` which uses both clusters and labelselector.
But I think the mwrs is better.